accounting/src/creditcommons/credit-commons-controller.ts

```diff
@@ -1,27 +1,47 @@
 import { AbstractCurrencyController } from "../controller/abstract-currency-controller"
 import { Context } from "../utils/context"
 import { CreditCommonsNode, CreditCommonsTransaction } from "../model/creditCommons"
-import { CreditCommonsController } from '../controller'
+import { LedgerCurrencyController } from '../controller/currency-controller'
 import { unauthorized } from "src/utils/error"
+import { TransferController } from "../controller/transfer-controller"
+import { InputTransfer } from "src/model/transfer"
+import { systemContext } from "src/utils/context"
 
+
+export interface CreditCommonsController {
+  getWelcome(ctx: Context): Promise<{ message: string }>
+  getAccountHistory(ctx: Context): Promise<{ data: object, meta: object }>
+  createNode(ctx: Context, ccNodeName: string, lastHash: string, vostroId: string): Promise<CreditCommonsNode>
+  createTransaction(ctx: Context, transaction: CreditCommonsTransaction): Promise<CreditCommonsTransaction>
+}
 export class CreditCommonsControllerImpl extends AbstractCurrencyController implements CreditCommonsController {
-  async createNode(ctx: Context, ccNodeName: string, lastHash: string): Promise<CreditCommonsNode> {
+  transferController: TransferController;
+  gatewayAccountId: string = '0';
+  ledgerScale: number  = 1000 * 1000;
+  ledgerBase: string = 'trunk/branch2/'
+  constructor(readonly currencyController: LedgerCurrencyController) {
+    super(currencyController)
+    this.transferController = new TransferController(currencyController)
```

Comment:
you can access the transfer from this class by calling `this.transfers()`, so no need to create a new TransferController.